### ISSUE TITLE
Stripping down js_optimize.py

### ISSUE THREAD

**Issue by waywardmonkeys** [kripken/emscripten#3695]
I'd like to remove some code paths from ``js_optimize.py`` so that we can report errors better / more correctly. This is in reaction to my having to spend some time debugging something where if ``js_optimize.py`` had failed more clearly, it would've been a lot easier.

* Do we need ``just_split``? I suspect that we do.
* Do we need ``know_generated``? I suspect that we don't. Removing this would let me handle an error situation a lot better (and would've saved me a good bit of time).

**Comment by kripken** [comment]
* Yes, I believe we use `just_split` that when receiving JSON instead of JS, or vice versa. It was added when we added JSON support.
 * Yes, I believe we always know generated. The only exception might be when we run `cleanup` which runs on the entire code, just parsing and unparsing it to clean it up. But I don't think that should stop you from removing `know_generated`, as we do the `cleanup` thing in a special way anyhow in that file.
